staging/src/k8s.io/apiserver/pkg/cel/library/cidr.go

```diff
@@ -229,7 +229,7 @@ func cidrContainsCIDR(arg ref.Val, other ref.Val) ref.Val {
 		return types.MaybeNoSuchOverloadErr(other)
 	}
 
-	equalMasked := cidr.Prefix.Masked() == netip.PrefixFrom(containsCIDR.Prefix.Addr(), cidr.Prefix.Bits())
+	equalMasked := cidr.Prefix.Masked() == netip.PrefixFrom(containsCIDR.Prefix.Addr(), cidr.Prefix.Bits()).Masked()
```

Comment:
yes, we need to compare the subnet id, that is equal to the masked ip address https://www.networkacademy.io/ccna/ip-subnetting/finding-the-subnet-id

Reply:
I had seen an alternative that was `cide.Contains` instead of overlaps here, do you think there's an advantage to this approach over the contains approach? I'd very much like to avoid false positives

Do you think these are easier to reason about than the current approach in the PR?

Reply:
it is a suggestion, binary arithmetic is always hard because is not how we use to reason about numbers :)

I personally find it easy to understand because how syntax is written: it overlaps , it has ips in common, and the network bits are lower (means is contained) and because the `equalMasked` variable is hard to parse, specially the second part of the condition, that has this pattern that is more common from other languages than golang

> equalMasked := cidr.Prefix.Masked() == netip.PrefixFrom(containsCIDR.Prefix.Addr(), cidr.Prefix.Bits()).Masked()

No strong opinion though, the tests cover the functionality, so up to you folks

Reply:
I had a look at how overlaps works
```
	// One of these Prefix calls might look redundant, but we don't require
	// that p and o values are normalized (via Prefix.Masked) first,
	// so the Prefix call on the one that's already minBits serves to zero
	// out any remaining bits in IP.
	var err error
	if p, err = p.ip.Prefix(minBits); err != nil {
		return false
	}
	if o, err = o.ip.Prefix(minBits); err != nil {
		return false
	}
	return p.ip == o.ip
```

It's pretty much identical to what I'm doing in the previous line, so, works for me

Reply:
it is the same, this is a good example of the difference of style I was referring in my original comment